### PR TITLE
Miscellaneous UI improvements and cleanup

### PR DESCRIPTION
**Pull request by m-rosario**
- Deleted the Components part of the constants library. I don't think this provided any value given that the components defined there were unique anyways.
- Deleted some obsolete Dimensions and Style constants.
- Deleted obsolete Widgets from previous design iterations, just to declutter the code. We can always go back to a previous version if for some reason we need to restore these.
- Replaced one of the mock network images being used, since it seems to no longer be available.
- Replaced the BottomNavigationBar with the [Material 3 NavigationBar](https://api.flutter.dev/flutter/material/NavigationBar-class.html).
- Some small aesthetic improvements on the Profile Quick View Card. Reduced padding below 'Recommended for you' and replaced left-right insets with start-end insets to support locales with right-to-left text. Made the highlight around the recommended card lighter and thinner:

image

### CODE REVIEW

**Review by jupblb — APPROVED** [review]

File: lib/widgets/molecules/recommended_mentors_scroll.dart
Comment (m-rosario) [reply lib/widgets/molecules/recommended_mentors_scroll.dart:21]: Oh I just read the URL haha

**Comment by m-rosario** [comment]
The approval was dismissed after I update the branch and resolved a conflict. Can I please get another approval?

**Review by jupblb — DISMISSED** [review]
We may reconsider forcing a PR to be 100% up to date before merging. The rationale for enabling this check was that if there are some tests for code dependent on something that may have changed in remote HEAD, then testing against outdated branch is not enough. Since we have no code coverage at this moment, I'm fine with disabling this. Let me know what you think. ;)

**Comment by m-rosario** [comment]
> We may reconsider forcing a PR to be 100% up to date before merging. The rationale for enabling this check was that if there are some tests for code dependent on something that may have changed in remote HEAD, then testing against outdated branch is not enough. Since we have no code coverage at this moment, I'm fine with disabling this. Let me know what you think. ;)

Having PR up to date before merging is probably better to avoid issues. Though, it is a annoying that the approval is dismissed after resolving a small conflict. But I guess maybe we just need to communicate better and be on top of these PRs.

**Review by ashlilili — APPROVED** [review]
LGTM!

**Pull request merged**
